resilience4j-ratpack/src/test/groovy/io/github/resilience4j/ratpack/circuitbreaker/CircuitBreakerTransformerSpec.groovy

```diff
@@ -93,53 +95,84 @@ class CircuitBreakerTransformerSpec extends Specification {
     def "can circuit break promise to from closed to open, then half open, then closed"() {
         given:
         CircuitBreaker breaker = buildBreaker()
-        CircuitBreakerTransformer<String> transformer = CircuitBreakerTransformer.of(breaker)
+        CircuitBreakerTransformer<String> circuitTransformer = CircuitBreakerTransformer.of(breaker)
         Exception e = new Exception("puke")
 
-        when:
+        when: "the first request and retry is a failure"
         def r = ExecHarness.yieldSingle {
             Blocking.<String> get { throw e }
-                    .transform(transformer)
+                    .transform(circuitTransformer)
+
         }
 
-        then:
+        then: "circuit is still closed since buffer hasn't been filled (1 call)"
         r.value == null
         r.error
         r.throwable == e
         breaker.state == CircuitBreaker.State.CLOSED
 
-        when:
+        when: "the request and the retry fails again"
         r = ExecHarness.yieldSingle {
             Blocking.<String> get { throw e }
-                    .transform(transformer)
+                    .transform(circuitTransformer)
         }
 
-        then:
+        then: "the circuit open since the buffer has at least 2 calls and the failure threshold was met"
         r.value == null
         r.error
         r.throwable == e
         breaker.state == CircuitBreaker.State.OPEN
 
-        when:
+        when: "a call is made after the wait duration that is a success"
         sleep 1000
         r = ExecHarness.yieldSingle {
             Blocking.<String> get { "foo" }
-                    .transform(transformer)
+                    .transform(circuitTransformer)
         }
 
-        then:
+        then: "the circuit is now in the HALF_OPEN state"
         r.value == "foo"
         !r.error
         r.throwable == null
         breaker.state == CircuitBreaker.State.HALF_OPEN
 
-        when:
+        and: "do a call where the Upstream promise handles the error, thus calling complete on the circuit"
+
+        when: "error on half open, retry swallows exception thrown"
+        r = ExecHarness.yieldSingle {
+            Blocking.<String> get { throw e }.onError { exception -> Promise.value("not foo") }
```

Comment:
Here's what it looks like to swallow the exception. If you dig into the Ratpack code if we handle an error and then produce a promised value, then all the downstreams will have a `complete` call propagated down effectively canceling them.